app/views/AutoTranslateView/index.tsx

```diff
@@ -1,162 +1,136 @@
+			if (letSelectedLanguage !== changes.autoTranslateLanguage) {
+				setSelectedLanguage(changes.autoTranslateLanguage);
+				letSelectedLanguage = changes.autoTranslateLanguage;
+			}
+			if (letAutoTranslate !== changes.autoTranslate) {
+				setEnableAutoTranslate(changes.autoTranslate);
+				letAutoTranslate = changes.autoTranslate;
+			}
+		});
+		return () => subscription.current?.unsubscribe && subscription.current.unsubscribe();
+	}, []);
+
+	const toggleAutoTranslate = async () => {
 		logEvent(events.AT_TOGGLE_TRANSLATE);
-		const { enableAutoTranslate } = this.state;
 		try {
+			setEnableAutoTranslate(!enableAutoTranslate);
 			await Services.saveAutoTranslate({
-				rid: this.rid,
+				rid,
 				field: 'autoTranslate',
 				value: enableAutoTranslate ? '0' : '1',
 				options: { defaultLanguage: 'en' }
 			});
-			this.setState({ enableAutoTranslate: !enableAutoTranslate });
 		} catch (error) {
+			setEnableAutoTranslate(!enableAutoTranslate);
 			logEvent(events.AT_TOGGLE_TRANSLATE_F);
-			console.log(error);
 		}
 	};
 
-	saveAutoTranslateLanguage = async (language: string) => {
+	const saveAutoTranslateLanguage = async (selectedLanguage: string) => {
 		logEvent(events.AT_SET_LANG);
 		try {
 			await Services.saveAutoTranslate({
-				rid: this.rid,
+				rid,
 				field: 'autoTranslateLanguage',
-				value: language
+				value: selectedLanguage
 			});
-			this.setState({ selectedLanguage: language });
+			setSelectedLanguage(selectedLanguage);
 		} catch (error) {
 			logEvent(events.AT_SET_LANG_F);
-			console.log(error);
 		}
 	};
 
-	renderIcon = () => {
-		const { theme } = this.props;
-		return <List.Icon name='check' color={themes[theme].tintColor} />;
-	};
-
-	renderSwitch = () => {
-		const { enableAutoTranslate } = this.state;
-		return <Switch value={enableAutoTranslate} trackColor={SWITCH_TRACK_COLOR} onValueChange={this.toggleAutoTranslate} />;
-	};
-
-	renderItem = ({ item }: { item: { language: string; name: string } }) => {
-		const { selectedLanguage } = this.state;
-		const { language, name } = item;
-		const isSelected = selectedLanguage === language;
-
-		return (
-			<List.Item
-				title={name || language}
-				onPress={() => this.saveAutoTranslateLanguage(language)}
-				testID={`auto-translate-view-${language}`}
-				right={() => (isSelected ? this.renderIcon() : null)}
-				translateTitle={false}
-			/>
-		);
-	};
-
-	render() {
-		const { languages } = this.state;
-		return (
-			<SafeAreaView testID='auto-translate-view'>
-				<StatusBar />
-				<List.Container testID='auto-translate-view-list'>
-					<List.Section>
+	const LanguageItem = React.memo(({ language, name }: { language: string; name?: string }) => (
+		<List.Item
+			title={name || language}
+			onPress={() => saveAutoTranslateLanguage(language)}
+			testID={`auto-translate-view-${language}`}
+			right={() => (selectedLanguage === language ? <List.Icon name='check' color={colors.tintColor} /> : null)}
+			translateTitle={false}
+		/>
+	));
```

Comment:
Can separate it in another file?

Reply:
There's even a way, but the logic of this component is tied to the logic of the local state, and as it's just a component, it's not at all complex to decide to keep it local.